src/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -85,6 +85,82 @@ public async Task Connect_AfterDisconnect_Fails()
                 }
             }
         }
+
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.OSX)] // Not supported on OSX.
+        public async Task ConnectGetsCanceledByDispose()
+        {
+            bool usesApm = UsesApm ||
+                           (this is ConnectTask); // .NET Core ConnectAsync Task API is implemented using Apm
+
+            // We try this a couple of times to deal with a timing race: if the Dispose happens
+            // before the operation is started, we won't see a SocketException.
+
+            SocketError? localSocketError = null;
+            bool disposedException = false;
+            for (int i = 0; i < 10 && !localSocketError.HasValue; i++)
+            {
+                var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+                Task connectTask = Task.Factory.StartNew(() =>
+                {
+                    ConnectAsync(client, new IPEndPoint(IPAddress.Parse("1.1.1.1"), 23)).GetAwaiter().GetResult();
+                }, TaskCreationOptions.LongRunning);
+
+                // Wait a little so the operation is started, then Dispose.
+                await Task.Delay(100);
```

Comment:
Why isn't it sufficient to do:
```C#
Task connectTask = ConnectAsync(...);
... // operation has started here, no delay needed
```
Is the additional delay necessary only for the synchronous case, where we're explicitly queueing a work item to start the operation?  And if so, could we add this delay only for the synchronous case?

Reply:
> Is the additional delay necessary only for the synchronous case, where we're explicitly queueing a work item to start the operation? 

Yes, we want the work item to be blocked.

> And if so, could we add this delay only for the synchronous case?

ok

Reply:
> And if so, could we add this delay only for the synchronous case?

This is also needed for the async case, otherwise those operations throw ODE and not the SE we are checking.